include/deal.II/base/vectorization.h

```diff
@@ -68,6 +68,27 @@ namespace std
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace internal
+{
+  template <typename T>
+  struct NumberType 
```

Comment:
We define a couple of number-related things in `include/deal.II/base/numbers.h`, so that would fit best in my opinion.